build/Build.CI.GitHubActions.cs

```diff
@@ -0,0 +1,96 @@
+﻿// Copyright 2021 Maintainers of NUKE.
+// Distributed under the MIT License.
+// https://github.com/nuke-build/nuke/blob/master/LICENSE
+
+using System;
+using System.Collections.Generic;
+using Nuke.Common.CI.GitHubActions;
+using Nuke.Common.CI.GitHubActions.Configuration;
+using Nuke.Common.Execution;
+using Nuke.Common.Utilities;
+
+[CustomGitHubActions("Build",
+    GitHubActionsImage.WindowsLatest,
+    GitHubActionsImage.UbuntuLatest,
+    AutoGenerate = true,
+    OnPushBranches = new[] { "master", "dev" },
+    OnPullRequestBranches = new[] { "master", "dev" },
+    CacheKeyFiles = new[] { "global.json", "src/**/*.csproj" },
+    InvokedTargets = new[] { nameof(Compile) },
+    OnPushExcludePaths = new[] { "docs/**/*", "package.json", "README.md" },
+    PublishArtifacts = false,
+    EnableGitHubContext = true)
+]
+
+[CustomGitHubActions("Tests",
+    GitHubActionsImage.WindowsLatest,
+    GitHubActionsImage.UbuntuLatest,
+    AutoGenerate = true,
+    OnPushBranches = new[] { "master", "dev" },
+    OnPullRequestBranches = new[] { "master", "dev" },
+    CacheKeyFiles = new[] { "global.json", "src/**/*.csproj" },
+    InvokedTargets = new[] { nameof(Test) },
+    OnPushExcludePaths = new[] { "docs/**/*", "package.json", "README.md" },
+    EnableGitHubContext = true)
+]
+
+partial class Build
+{
+}
+class CustomGitHubActionsAttribute : GitHubActionsAttribute
+{
+    public CustomGitHubActionsAttribute(string name, GitHubActionsImage image, params GitHubActionsImage[] images) : base(name, image, images)
+    {
+    }
+
+    protected override GitHubActionsJob GetJobs(GitHubActionsImage image, IReadOnlyCollection<ExecutableTarget> relevantTargets)
+    {
+        GitHubActionsJob job = null;
+        try
+        {
+            job = base.GetJobs(image, relevantTargets);
+            var newSteps = new List<GitHubActionsStep>(job.Steps);
+            foreach (var version in new[] { "6.0.*", "5.0.*" })
+            {
+                newSteps.Insert(1, new GitHubActionsSetupDotNetStep
+                {
+                    Version = version
+                });
+            }
+
+            job.Steps = newSteps.ToArray();
+            return job;
+        }
+        catch (Exception ex)
+        {
+
+        }
+        return job;
+    }
+}
+
+class GitHubActionsSetupDotNetStep : GitHubActionsStep
```

Comment:
Installing SDKs with job not currently supported in Nuke, hence the need for the custom github action attribute